trino/dbapi.py

```diff
@@ -201,6 +204,31 @@ def __init__(
         self.legacy_primitive_types = legacy_primitive_types
         self.legacy_prepared_statements = legacy_prepared_statements
 
+        if not defer_connect:
```

Comment:
```suggestion
        if not defer_connect and auth != constants.DEFAULT_AUTH:
```

@hovaesco This would mean we'll only do eager connection if some authentication is actually provided. WDYT? Or do you think existing code is better (simpler and we don't assume when someone might want eager connections)?

Reply:
I don't see any production use cases when auth is `constants.DEFAULT_AUTH` but might be helpful in some local testing.